scipy/special/boost_special_functions.h

```diff
@@ -2243,4 +2243,54 @@ f_sf_double(double dfn, double dfd, double x)
 {
     return f_sf_wrap(dfn, dfd, x);
 }
+
+template<typename Real>
+Real
+chdtriv_wrap(const Real p, const Real x)
+{
+    if (std::isnan(p) || std::isnan(x)) {
+        return NAN;
+    }
+    if ((x <= 0) || (p < 0) || (p > 1)) {
+        sf_error("chdtriv", SF_ERROR_DOMAIN, NULL);
+        return NAN;
+    }
+    if ((p == 1)) {
+        return 0.0;
+    }
+    Real y;
+    try {
+        y = 2 * boost::math::gamma_p_inva<Real>(0.5 * x, p, SpecialPolicy());
+    } catch (const std::domain_error& e) {
+        sf_error("chdtriv", SF_ERROR_DOMAIN, NULL);
+        y = NAN;
+    } catch (const std::overflow_error& e) {
+        sf_error("chdtriv", SF_ERROR_OVERFLOW, NULL);
+        y = INFINITY;
+    } catch (const std::underflow_error& e) {
+        sf_error("chdtriv", SF_ERROR_UNDERFLOW, NULL);
+        y = 0;
+    } catch (...) {
+        sf_error("chdtriv", SF_ERROR_NO_RESULT, NULL);
+        y = NAN;
+    }
+    if (y < 0) {
+        sf_error("chdtriv", SF_ERROR_NO_RESULT, NULL);
+        y = NAN;
```

Comment:
Sorry, one more question: can this case occur? Can gamma_p_inv() return a negative value?

Reply:
I do that check now for all boost wrappers, just to be on the safe side as we have seen that boost can return values outside $[0, 1]$ for different CDFs. Compared to the actual computation, this check is very cheap so I think it does not hurt.

Reply:
OK.  I think I've done the same thing in the past.